Feature/doc restructure rclc (#13)
microROS/micro-ROS-doc: rclc/API_guide.md

# API guide

## rclc

```C
rclc_ret_t rclc_init(int argc, char const * const * argv)
```

> Function that initialize all micro-ROS stack in a cascade way, initialize rcl and rcl initialize rmw.

Reviewer: ```suggestion
A function that initialises all micro-ROS stack in a cascade way, it initialises rcl and rcl initialises rmw.
```

Arguments are passed down to the rcl layer.
This function should be called before any other one.

`argc`
Init args count.

`argv`
Init args pointer.

`return`
[rcl response](http://docs.ros2.org/beta1/api/rcl/types_8h.html)

---
```C
bool rclc_ok(void)
```
Checks the status of rcl initialization.

`return`
True if OK.

---

## Node

```C
rclc_node_t * rclc_create_node(const char * name, const char * namespace_)
```
Creates a ROS node.
It initializes lower layer nodes, rcl_node and rmw_node.

`name`
Pointer to the node name string.

`namespace_`

> Pointer to the name space string.

Reviewer: ```suggestion
Pointer to the namespace string.
```

`return`

---
```C
rclc_ret_t rclc_destroy_node(rclc_node_t * node)
```
Removes a ROS node and all its related publishers and subscribers.

`node`

> Pointer to owner to be destroyed.

Reviewer: ```suggestion
The pointer to the node to be destroyed.
```

`return`
[rcl response](http://docs.ros2.org/beta1/api/rcl/types_8h.html)
---
```C
void rclc_spin_node_once(rclc_node_t * node, int64_t time_out_ms)
```
This function listens to the ROS subscriptions from the ROS node for a given time.
It triggers user callback calls with ROS messages received on subscriptions.

`node`
Pointer to the node to spin.

`time_out_ms`
Spin timeout.

```C
void rclc_spin_node(rclc_node_t * node)
```

> Listens to the ROS subscriptions from the ROS node till the user close the application.

Reviewer: ```suggestion
Listens to the ROS subscriptions from the ROS node until the user closes the application.
```

It triggers user callback calls with ROS messages received on subscriptions.

`node`
Pointer to the node to spin.

---

## Publisher

```C
rclc_publisher_t * rclc_create_publisher(rclc_node_t * node, const rclc_message_type_support_t type_support, const char * topic_name, size_t queue_size)
```
The rclc_create_publisher function creates a ROS publisher within the given node.
The created publisher publish on the ROS topic `topic_name` using `type_support` as the type for the ROS messages.

`node`
Pointer to the owner node.

`type_support`
Pointer to the message type support.

`topic_name`
Pointer to the topic name string.

`queue_size`
Queue size.

`return`
Pointer to the created publisher.

---
```C
rclc_ret_t rclc_destroy_publisher(rclc_publisher_t * publisher)
```
Removes the publisher.

`publisher`
Pointer to the publisher to be destroyed.

`return`
[rcl response](http://docs.ros2.org/beta1/api/rcl/types_8h.html)

---
```C
rclc_ret_t rclc_publish(const rclc_publisher_t * publisher, const void * ros_message)
```
Publish a ROS message using the ROS topic tied to the `publisher`.

`publisher`
Pointer to the publisher.

`ros_message`

> Pointer to the ros message.

Reviewer: ```suggestion
Pointer to the ROS message.
```

`return`
[rcl response](http://docs.ros2.org/beta1/api/rcl/types_8h.html)

## Subscription

```C
rclc_subscription_t * rclc_create_subscription(rclc_node_t * node, const rclc_message_type_support_t type_support, const char * topic_name, rclc_callback_t callback, size_t queue_size,bool ignore_local_publications)
```
Creates a ROS subscription within the given node.

> The created subscription attends on the ROS topic `topic_name` using `type_support` as the type for the ROS messages.

Reviewer: ```suggestion
The created subscription listens on the ROS topic `topic_name` using `type_support` as the type for the ROS messages.
```
